src/elf/mod.rs

```diff
@@ -379,7 +402,7 @@ if_sylvan! {
         }
     }
 
-    fn hash_len(bytes: &[u8], offset: usize, machine: u16, ctx: Ctx) -> error::Result<usize> {
+    pub fn hash_len(bytes: &[u8], offset: usize, machine: u16, ctx: Ctx) -> error::Result<usize> {
```

Comment:
ditto

Reply:
this also feels very lowlevel as a pub function; on argument is a `u16` machine value, which doesn't seem great for user to think about; if this is needed for some reason, i think we can provide something better here (still not sure why it's needed exactly, hoping you can fill that in)